presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/security/FileBasedAccessControl.java

```diff
@@ -211,10 +212,28 @@ public void checkCanShowColumns(ConnectorSecurityContext context, SchemaTableNam
     @Override
     public List<ColumnMetadata> filterColumns(ConnectorSecurityContext context, SchemaTableName tableName, List<ColumnMetadata> columns)
     {
-        if (!checkAnyTablePermission(context, tableName)) {
+        if (INFORMATION_SCHEMA_NAME.equals(tableName.getSchemaName())) {
+            return columns;
+        }
+
+        ConnectorIdentity identity = context.getIdentity();
+        TableAccessControlRule rule = tableRules.stream()
+                .filter(tableRule -> tableRule.matches(identity.getUser(), identity.getGroups(), tableName))
+                .findFirst()
+                .orElse(null);
+        if (rule == null || rule.getPrivileges().isEmpty()) {
             return ImmutableList.of();
         }
-        return columns;
+
+        // if user has privileges other than select, show all columns
```

Comment:
Why?

Reply:
Currently column permissions only apply to SELECT.  If you have insert permissions you can insert into any column, so it makes sense that you can see these columns in a show statement (or information schema).